server/accounts.go

```diff
@@ -1343,6 +1343,14 @@ func (a *Account) AddServiceImportWithClaim(destination *Account, from, to strin
 		return ErrServiceImportAuthorization
 	}
 
+	// Check that what we are importing is not something we also export.
+	if a.serviceExportExists(to) {
+		// So at this point if destination account is also importing from us, that forms a cycle.
```

Comment:
Would this also happen if the cycle would be bigger, A - B -C - A?

Reply:
You raise a good point. I have to detect cycles at runtime. I will make it similar to routes in that once you are mapped via a service import you can't be mapped again. Will keep what we have but do a test like above.